Tests/NIOTests/ChannelTests.swift

```diff
@@ -2200,4 +2200,108 @@ public class ChannelTests: XCTestCase {
+                hasClosed = true
+            }
+            XCTAssertNoThrow(try serverSockSyncQ.sync { try serverSock.close() })
+        }
+        let g = DispatchGroup()
+        let q = DispatchQueue(label: "q")
+        do {
+            // spawn 64 connects, likely some will already get ECONNRESET as the backlog isn't large enough (works quite well on Linux, not so much on Darwin)
+            let fs = (0..<64).map { _ in ClientBootstrap(group: group).connect(to: serverSockAddress) }
+            q.async(group: g) { () -> Void in
+                while true {
+                    do {
+                        // try to accept one client
+                        let accepted = try serverSockSyncQ.sync { () -> CInt? in
+                            try serverSock.withUnsafeFileDescriptor { fd in
+                                var pfd = pollfd(fd: fd, events: Int16(POLLIN | POLLERR | POLLHUP), revents: 0)
+                                while true {
+                                    // wait a maximum of 500ms to accept a new client, just to be guaranteed we don't get stuck here.
+                                    switch poll(&pfd, 1, 500) {
+                                    case 1:
+                                        return accept(fd, nil, nil)
+                                    case 0:
+                                        // spurious wakeup
+                                        continue
+                                    case let ret:
+                                        let saveErrno = errno // let's try our best
+                                        XCTAssertEqual(-1, ret)
+                                        XCTAssertTrue(hasClosed, "error \(saveErrno) happened (after poll returned \(ret)) but we haven't closed yet")
+                                        return nil
+                                    }
+                                }
+                            }
+                        }
+                        if let accepted = accepted {
+                            // set SO_LINGER to 0, which will cause most TCP stacks to send an RST packet (works quite well on Darwin, not so much on Linux)
+                            var lin = linger(l_onoff: 1, l_linger: 0)
+                            setsockopt(accepted, SOL_SOCKET, SO_LINGER, &lin, socklen_t(MemoryLayout.size(ofValue: lin)))
+                            // immediately close the accepted socket
+                            close(accepted)
+                        } else {
+                            break
+                        }
+                    } catch let error as IOError where error.errnoCode == EBADF {
+                        // totally fine as we just close the server socket from the main thread if we have seen
+                        // the error we're waiting for.
+                        break
+                    } catch {
+                        XCTFail("unexpected error \(error)")
+                    }
```

Comment:
Here's my question: is there any reason we can't do this with a regular server socket channel? Create it, set the child channel options, turn off auto-read so that we only accept a maximum of one.

Reply:
@Lukasa I actually will scrap this whole test. It just isn't causing the `ECONNRESET` reliably enough on all the platforms and there's always room for random failure of the test (unless I don't assert that at least one actually gets `ECONNRESET`). But tbh I think pretty much everything is covered by the `ECONNREFUSED` test already...